flocker/node/_deploy.py

```diff
@@ -674,6 +690,7 @@ def applications_from_units(units):
                 ))
 
             return [NodeState(
+                uuid=self.uuid,
```

Comment:
I did however previously deliberately port application deployer discovery tests to new model, so I think coverage is sufficient.